silencer-plugin/src/main/scala/com/github/ghik/silencer/SilencerPlugin.scala

```diff
@@ -11,26 +12,43 @@ class SilencerPlugin(val global: Global) extends Plugin { plugin =>
   val description = "Scala compiler plugin for warning suppression"
   val components: List[PluginComponent] = List(component)
   private var globalFilters = List.empty[Regex]
+  private var pathFilters = List.empty[Regex]
+  private var sourceRoots = List.empty[File]
 
   private lazy val reporter =
-    new SuppressingReporter(global.reporter, globalFilters)
+    new SuppressingReporter(global.reporter, globalFilters, pathFilters, sourceRoots)
 
-  override def processOptions(options: List[String], error: String => Unit): Unit = {
-    options.foreach { opt =>
-      if (opt startsWith "globalFilters=") {
-        globalFilters = opt.drop(14).split(";").map(_.r).toList
-      }
-    }
+  private def split(s: String) = s.drop(1).split(";")
 
-    if (globalFilters.nonEmpty) {
-      global.inform(s"""Silencer using global filters: ${globalFilters.mkString(",")}""")
-    }
+  override def processOptions(options: List[String], error: String => Unit): Unit = {
+    options.foreach(_.span(_ > '=') match {
+      case ("globalFilters", pattern) =>
+        globalFilters = globalFilters ++ split(pattern).map(_.r)
+      case ("pathFilters", pattern) =>
+        pathFilters = pathFilters ++ split(pattern).map(_.r)
+      case ("sourceRoots", rootPaths) =>
+        sourceRoots = sourceRoots ++ split(rootPaths).map(new File(_))
+      case (unsupportedOption, _) =>
+        global.inform(s"Silencer does not support filter option: $unsupportedOption")
+    })
+
+    if (globalFilters.nonEmpty)
```

Comment:
TBH these messages are only noise. IMHO they should all be removed (along with the original one for `globalFilters`)

Reply:
removed them, I found them to be just noise as well